services/src/main/java/org/keycloak/authentication/AuthenticationProcessor.java

```diff
@@ -854,6 +855,9 @@ public void validateUser(UserModel authenticatedUser) {
         if (authenticatedUser == null) return;
         if (!authenticatedUser.isEnabled()) throw new AuthenticationFlowException(AuthenticationFlowError.USER_DISABLED);
         if (realm.isBruteForceProtected()) {
+            if (event.getEvent().getType().equals(EventType.RESET_PASSWORD)) {
```

Comment:
This is not a good approach. You can't rely on the even type. It's only there to be able to send events, not as a way to make authentication logic.